subprojects/parseq/src/test-utils/java/com/linkedin/parseq/junitjupiter/BaseEngineJUnitJupiterTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2019 LinkedIn, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.linkedin.parseq.junitjupiter;
+
+import com.linkedin.parseq.AbstractBaseEngineTest;
+import com.linkedin.parseq.EngineBuilder;
+import com.linkedin.parseq.ParSeqUnitTestHelper;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+
+
+/**
+ * A base class that builds an Engine with default configuration.
+ * Intended to be used with JUnit Jupiter (JUnit5+)
+ *
+ * This class creates new Engine and shuts it down before and after every test method, so it can't be used
+ * to run tests in parallel.
+ *
+ * The difference between this class and {@link BaseEngineParJunitJupiterTest} is that {@code BaseEngineParJunitJupiterTest} creates new
+ * {@code Engine} instance only once for all tests in the class and thus can be used to run test methods in parallel.
+ *
+ * @see ParSeqUnitTestHelper
+ * @see BaseEngineParJunitJupiterTest
+ */
+public class BaseEngineJUnitJupiterTest extends AbstractBaseEngineTest {
```

Comment:
I myself is also debating whether we should use a different class name or the same name since you have put that in a different package. The downside for this is that when users remove `testNg` dependency, they have to not only change imports but also code. The pro is that this will be very explicit for users to think what class to use.

Reply:
I suggest an explicit, different name. The migration argument is not strong enough: search replace across files is easy.

Happy to change the name to be the same if that's the team's decision. I don't mind either way.

Reply:
Ok, you convinced me.